doc/api/readline.md

```diff
@@ -88,6 +88,28 @@ rl.on('line', (input) => {
 });
 ```
 
+### Event: `'history'`
+<!-- YAML
+added: REPLACEME
+-->
+
+The `'history'` event is emitted whenever the history array has changed.
+
+The listener function is called with an array containing the history array.
+It will reflect all changes, added lines and removed lines due to
+`historySize` and `removeHistoryDuplicates`.
+
+The primary purpose is to allow a listener to persist the history.
+It is also possible for the listener to change the history object. This
+could be useful to prevent certain lines to be added to the history, like
+a password.
```

Comment:
I just noticed this change while working on readline.

This is a pretty bad idea in my opinion. The user should never manipulate the history on their own. If we want to prevent addition of entries, I would rather have a method that received the line and the return decides if it is added to the history or not. That way the same could be achieved without ever being added to the history array. Otherwise, I would be able to add another listener that I push to the front of the 'history' listeners and just safe the value there.

Changing the history can also always have unforeseen side-effects. Ideally, our history is immutable, as long as not specific methods are used to mutate these entries in a controlled fashion.